src/main/java/uk/ac/ebi/embl/template/TemplateEntryProcessor.java

```diff
@@ -378,4 +395,122 @@ private void validateMarker(TemplateProcessorResultSet templateProcessorResultSe
             templateProcessorResultSet.getValidationResult().append(new ValidationResult().append(message));
         }
     }
+
+    /**
+     * This method check if the ORGANISM_NAME field value is related to a sample and 
+     * updates the entry's sourceFeature using the sample values.
+     * 
+     * @param entry
+     * @param templateVariables
+     * @param templateProcessorResultSet
+     * @param options
+     * @throws Exception
+     */
+    private void updateSourceFeatureUsingOrganismFieldValue(Entry entry, TemplateVariables templateVariables, TemplateProcessorResultSet templateProcessorResultSet, SubmissionOptions options) throws Exception{
+
+        // Validate and get samples
+        Sample sample = validateAndGetSample(templateVariables, templateProcessorResultSet,options);
+
+        if(sample != null && entry.getPrimarySourceFeature()!=null){
+            SourceFeature sourceFeature=entry.getPrimarySourceFeature();
+            SampleInfo sampleInfo=getSampleInfo(sample);
+            SampleEntity sampleEntity = getSampleEntity(sample);
+            updateSourceFeature(sourceFeature,sampleEntity,sampleInfo);
+            entry.addXRef(new XRef("BioSample", sample.getBioSampleId()));
+        }
+    }
+
+    /**
```

Comment:
/**
Returns sample if one could be found using the ORGANISM_NAME field of the TSV file or NULL if the ORGANISM_NAME was a scientific name or tax id. 
*/